care/emr/api/viewsets/report/report_upload.py

```
@@ -148,6 +152,53 @@ def generate(self, request, *args, **kwargs):
             status=status.HTTP_201_CREATED,
         )
 
+    @extend_schema(
+        request=GenerateReportRequest, responses={200: "Success"}, tags=["report"]
+    )
+    @action(detail=False, methods=["POST"])
+    def preview(self, request, *args, **kwargs):
+        if not AuthorizationController.call("can_preview_template", request.user):
+            raise PermissionDenied("You dont have permission to preview reports")
```

Ruff (0.15.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)